waku/waku_api/rest/server.nim

```diff
@@ -46,17 +58,68 @@ proc default*(T: type RestServerConf): T =
 
 ### Initialization
 
-proc getRouter(allowedOrigin: Option[string]): RestRouter =
+proc new*(t: typedesc[WakuRestServerRef],
+          router: RestRouter,
+          address: TransportAddress,
+          serverIdent: string = PrestoIdent,
+          serverFlags = {HttpServerFlags.NotifyDisconnect},
+          socketFlags: set[ServerFlags] = {ReuseAddr},
+          serverUri = Uri(),
+          maxConnections: int = -1,
+          backlogSize: int = DefaultBacklogSize,
+          bufferSize: int = 4096,
+          httpHeadersTimeout = 10.seconds,
+          maxHeadersSize: int = 8192,
+          maxRequestBodySize: int = 1_048_576,
+          requestErrorHandler: RestRequestErrorHandler = nil,
+          dualstack = DualStackType.Auto,
+          allowedOrigin: Option[string] = none(string)
+          ): RestServerResult[WakuRestServerRef] =
+  var server = WakuRestServerRef(router: router)
+
+  let restMiddleware = RestServerMiddlewareRef.new(router = server.router, errorHandler = requestErrorHandler)
+  let originHandlerMiddleware = OriginHandlerMiddlewareRef.new(allowedOrigin)
+
+  let middlewares = [originHandlerMiddleware,
+                     restMiddleware]
+
+  proc defaultProcessCallback(rf: RequestFence): Future[HttpResponseRef] {.
+        async: (raises: [CancelledError]).} =
+    discard
```

Comment:
Is this doing nothing because we are passing `middlewares`? In that case I think is important to add a comment in this proc to clarify that